src/plugins/intel_cpu/src/nodes/executors/subgraph.hpp

```diff
@@ -98,10 +99,42 @@ class SubgraphBaseExecutor {
     std::vector<ptrdiff_t> m_start_offset_out = {};
 };
 
+class SubgraphSpecializedBaseExecutor {
+public:
+    SubgraphSpecializedBaseExecutor(const std::set<size_t>& external_ptrs_idces, size_t in_num) {
+        precompute_ptr_mappings(external_ptrs_idces, in_num, m_src_ptr_mappings, m_external_ptr_mappings);
+    };
+
+protected:
+    struct PtrMapping {
+        size_t original_idx;
+        size_t postprocessed_idx;
+    };
+
+    inline void precompute_ptr_mappings(const std::set<size_t>& external_ptrs_idces,
+                                        size_t total_size,
+                                        std::vector<PtrMapping>& src_ptr_mappings,
+                                        std::vector<PtrMapping>& external_ptr_mappings) {
+        size_t external_idx = 0, src_idx = 0;
+        for (size_t i = 0; i < total_size; i++) {
```

Comment:
By the way, could we allocate memory firstly?
```
external_ptr_mappings.resolve(external_ptrs_idces.count());
src_ptr_mappings.resolve(total_size - external_ptrs_idces.count());
```

Reply:
Also I think this method should be private - it should be called only once in ctor of base class. And since this function is method of the class, we can:
- remove method and move its body to the ctor - we need it only in ctor and don't call this method anywhere
- or remove args `src_ptr_mappings` and `external_ptr_mappings` - we can work with the class fields :)

What do you think about it? I'd prefer first option and I believe it can help us to simplify this already simple class :))

Reply:
> By the way, could we allocate memory firstly?
> 
> ```
> external_ptr_mappings.resolve(external_ptrs_idces.count());
> src_ptr_mappings.resolve(total_size - external_ptrs_idces.count());
> ```

Applied, thank you!



> Also I think this method should be private - it should be called only once in ctor of base class. And since this function is method of the class, we can:
> 
> * remove method and move its body to the ctor - we need it only in ctor and don't call this method anywhere
> * or remove args `src_ptr_mappings` and `external_ptr_mappings` - we can work with the class fields :)
> 
> What do you think about it? I'd prefer first option and I believe it can help us to simplify this already simple class :))

I like 1st option more, implemented, please take a look